free/src/main/scala/cats/free/Free.scala

```diff
@@ -80,8 +81,8 @@ sealed abstract class Free[S[_], A] extends Product with Serializable {
    */
   final def runM[M[_]](f: S[Free[S, A]] => M[Free[S, A]])(implicit S: Functor[S], M: Monad[M]): M[A] = {
     def runM2(t: Free[S, A]): M[A] = t.resume match {
-      case Left(s) => Monad[M].flatMap(f(s))(runM2)
-      case Right(r) => Monad[M].pure(r)
+      case Xor.Left(s) => Monad[M].flatMap(f(s))(runM2)
```

Comment:
Yeah it is - I just went around changing stuff from `Either` to `Xor`, didn't really look at implementations at all.

I think I know how to change this to use `tailRecM` but since you and Eric are currently reviewing and I'd like to keep this all in one commit, I'll let you guys finish reviewing, fix it, have you give it a glance, and push.